tests/fixtures/endpoints_handler.py

```diff
@@ -0,0 +1,206 @@
+            key=lambda endpoint: len(endpoint.url_pattern), reverse=True
+        )
+
+    @classmethod
+    def get(cls, url: str):
+        """Decorator function to add endpoint to Handler"""
+
+        def inner(endpoint_function: Callable):
+            cls._add_endpoint(
+                method="GET", url=url, endpoint_function=endpoint_function
+            )
+            return endpoint_function
+
+        return inner
+
+    @classmethod
+    def post(cls, url: str):
+        """Decorator function to add endpoint to Handler"""
+
+        def inner(endpoint_function: Callable):
+            cls._add_endpoint(
+                method="POST", url=url, endpoint_function=endpoint_function
+            )
+            return endpoint_function
+
+        return inner
+
+    @classmethod
+    def patch(cls, url: str):
+        """Decorator function to add endpoint to Handler"""
+
+        def inner(endpoint_function: Callable):
+            cls._add_endpoint(
+                method="PATCH", url=url, endpoint_function=endpoint_function
+            )
+            return endpoint_function
+
+        return inner
+
+    @staticmethod
+    def convert_parameter_types(
+        endpoint_function: Callable,
+        string_parameters: dict[str, str],
+        request: httpx.Request,
+    ) -> dict[str, Any]:
+        """Inspect the signature of the endpoint function and determine the parameter
+        types. Since the values parsed from the URL are still in string format, cast
+        them to the types specified in the signature. If the request is needed, include
+        that in the returned parameters"""
+
+        # Get the parameter information from the endpoint function signature
+        signature_parameters = inspect.signature(endpoint_function).parameters
+
+        # type-cast based on type-hinting info
+        typed_parameters: dict[str, Any] = {}
+        for parameter_name, value in string_parameters.items():
+            parameter_type = signature_parameters[parameter_name].annotation
+
+            # all parameters should be typed, raise exception otherwise
+            if parameter_type is inspect.Parameter.empty:
+                raise Exception(
+                    f"Parameter '{parameter_name}' in function "
+                    + f"'{endpoint_function.__name__}' is missing type information!"
+                )  # pylint: disable=broad-exception-raised
+
+            if parameter_type is not str:
+                value = parameter_type(value)
+            typed_parameters[parameter_name] = value
+
+        # include request itself if needed (e.g. for header or auth info),
+        if "request" in signature_parameters:
+            typed_parameters["request"] = request
+
+        logger.info("\tParameters are:")
+        for name, value in typed_parameters.items():
+            logger.info("\t\t%s: %s (%s)", name, value, type(value))
+
+        return typed_parameters
+
+    @classmethod
+    def get_function_and_parameters(
```

Comment:
```suggestion
    def _get_function_and_parameters(
```

Reply:
Changed in [f9a5cf1](https://github.com/ghga-de/ghga-connector/pull/68/commits/f9a5cf199803a93f4f8f26826410e0c883a01d7b)